make/common/JdkNativeCompilation.gmk

```diff
@@ -76,18 +76,161 @@ endif
+        $1_$2_JVM_VARIANT_PATH := server
+      else
+        $1_$2_JVM_VARIANT_PATH := $(JVM_VARIANT_MAIN)
+      endif
+      ifeq ($$(call isTargetOs, windows), true)
+        ifeq ($(STATIC_LIBS), true)
+          $1_$2_LIBPATH := $$(HOTSPOT_OUTPUTDIR)/variant-$$($1_$2_JVM_VARIANT_PATH)/libjvm/objs/static
+        else
+          $1_$2_LIBPATH := $$(SUPPORT_OUTPUTDIR)/modules_libs/$$($1_$2_MODULE)
+        endif
+      else
+        ifeq ($(STATIC_LIBS), true)
+          $1_$2_LIBPATH := $$(HOTSPOT_OUTPUTDIR)/variant-$$($1_$2_JVM_VARIANT_PATH)/libjvm/objs/static
+        else
+          $1_$2_LIBPATH := $$(SUPPORT_OUTPUTDIR)/modules_libs/$$($1_$2_MODULE)/$$($1_$2_JVM_VARIANT_PATH)
+        endif
+      endif
+    endif
+  else
+    # Special treatment for virtual module "gtest"
+    ifeq ($$($1_$2_NAME), jvm)
+      ifeq ($$(call isTargetOs, windows), true)
+        ifeq ($(STATIC_LIBS), true)
+          $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/gtest/objs/static
+        else
+          $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/gtest/objs
+        endif
+      else
+        ifeq ($(STATIC_LIBS), true)
+          $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/gtest/objs/static
+        else
+          $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/gtest
+        endif
+      endif
+    else ifeq ($$($1_$2_NAME), gtest)
+      $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/libgtest
+      ifeq ($(STATIC_LIBS), true)
+        $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/libgtest/objs/static
+      else
+        $1_$2_LIBPATH := $$(JVM_OUTPUTDIR)/libgtest
+      endif
+    endif
+  endif
+endef
+
+# Process a JDK library designation such as "java.desktop:jawt" into the needed
+# additions to EXTRA_LINK_DEPS, LDFLAGS, LIBS and EXTRA_HEADER_DIRS.
+# $1: The name of the rule (namespace)
+# $2: The safe namespace of the library
+# $3: The actual string given to represent the library
+define AddJdkLibrary
+  $1_$2_NAME := $$(strip $$(lastword $$(subst :, , $3)))
+  $1_$2_MODULE := $$(strip $$(patsubst %$$($1_$2_NAME), %, $3))
+
+  ifeq ($$(filter lib%, $$($1_$2_NAME)),)
+    $$(error Library name $$($1_$2_NAME) missing lib prefix in $1)
+  endif
+  $1_$2_NAME := $$(strip $$(patsubst lib%, %, $$($1_$2_NAME)))
+
+  ifeq ($$($1_$2_MODULE),)
+    $1_$2_MODULE := $$(MODULE)
+  else
+    $1_$2_MODULE := $$(strip $$(patsubst %:, %, $$($1_$2_MODULE)))
+  endif
+
+  # Determine if the library in question is static.
+  ifeq ($(STATIC_BUILD), true)
+     $1_$2_STATIC_LIBRARY := true
+  endif
+  # Ideally, we should not hardcode these
+  ifeq ($(call isTargetOs, aix)+$$($1_$2_MODULE):$$($1_$2_NAME), true+java.base:jli)
+     $1_$2_STATIC_LIBRARY := true
+  endif
+  ifeq ($$($1_$2_MODULE):$$($1_$2_NAME), gtest:gtest)
+     $1_$2_STATIC_LIBRARY := true
+  endif
+
+  # Setup $1_$2_LIBPATH.
+  $$(eval $$(call ResolveLibPath,$1,$2))
+
+  $1_EXTRA_HEADER_DIRS += $$($1_$2_MODULE):lib$$($1_$2_NAME)
+
+  ifneq ($(STATIC_LIBS), true)
+    ifeq ($$(call isTargetOs, windows), true)
```

Comment:
I should've looked through this more carefully. The selection on whether to use \-L and \-libpath: should've been based on the compiler, not the target OS